### PR TITLE
Revert "Revert "Fix dsl i18n keys""

### PR DESCRIPTION
**Pull request by Hamms**
Reverts code-dot-org/code-dot-org#29148

Fixes the bug (which Bethany correctly identified in the COE doc) and adds a couple UI tests

### CODE REVIEW

File: dashboard/app/views/levels/_single_multi.html.haml [hunk 1]
```diff
@@ -21,7 +21,7 @@
   = render partial: 'levels/content', locals: {app: app, data: data, content_class: question_content_class, source_level: level, hide_header: !standalone}
 
   - localized_questions = level.localized_property(:questions)
-  - if localized_questions.count
+  - if localized_questions.any?
```
Comment (davidsbailey) [inline dashboard/app/views/levels/_single_multi.html.haml:24]: 🤦‍♂ this mistake would be harder to make if ruby and javascript would agree on whether `0` is falsy

File: dashboard/test/ui/features/multi.feature [hunk 2]
```diff
@@ -30,8 +30,21 @@ Scenario: Submitting an incorrect option
   And I press ".modal #ok-button" using jQuery
   And I wait until element "#cross_0" is visible
 
+Scenario: Rendering in another language
+  Given I am on "http://studio.code.org/s/course1/stage/2/puzzle/2/lang/es-MX"
+  Then I rotate to landscape
+  And I wait to see ".submitButton"
+  And element ".submitButton" is visible
+  Then element ".multi h1" has "es-MX" text from key "data.multi.2-3 Algorithms Multi 1.title"
```
Comment (davidsbailey) [inline dashboard/test/ui/features/multi.feature:38]: very cool!

**Review by davidsbailey — APPROVED** [review]

File: dashboard/app/views/levels/_single_multi.html.haml (hunk 1 above)
Comment (Hamms) [reply dashboard/app/views/levels/_single_multi.html.haml:24]: right?!

**Review by bethanyaconnor — APPROVED** [review]
I assume the added test represents the production issue we saw. Are we worried about any other unexpected cases cropping up?

**Comment by Hamms** [comment]
It does, and I'm not. We've actually got pretty good testing for DSL content (including _great_ tests for things like https://studio.code.org/s/allthethings/stage/23/puzzle/2/page/1 that contain just a ton of different DSL stuff); this one edge case just managed to slip through

**Pull request merged**